sklearn/metrics/tests/test_common.py

```diff
@@ -1942,6 +1942,7 @@ def check_array_api_metric_pairwise(metric, array_namespace, device, dtype_name)
         check_array_api_multiclass_classification_metric,
     ],
     mean_tweedie_deviance: [check_array_api_regression_metric],
+    partial(mean_tweedie_deviance, power=-0.5): [check_array_api_regression_metric],
```

Comment:
```suggestion
    partial(mean_tweedie_deviance, power=-0.5): [check_array_api_regression_metric],
    partial(mean_tweedie_deviance, power=1.5): [check_array_api_regression_metric],
```

`mean_poisson_deviance` and `mean_gamma_deviance` will or should already cover the other cases.

Reply:
Thank you! Though power=1.5 is a case for `mean_poisson_deviance` called `mean_compound_poisson_deviance` ([test case here](https://github.com/scikit-learn/scikit-learn/blob/97f581825e7eb0a15c02825d9da0731e491c9861/sklearn/metrics/tests/test_common.py#L123), Poisson's condition is 1 <= power < 2) and inputs within the range of (0.0, 1.0) is also invalid in general for `mean_tweedie_deviance` (validate params check [here](https://github.com/scikit-learn/scikit-learn/blob/97f581825e7eb0a15c02825d9da0731e491c9861/sklearn/metrics/_regression.py#L1334-L1337)) . So I actually think power=-0.5 is the only case needed here, but please let me know if I have missed anything, thanks! :)

Reply:
`mean_poisson_deviance` is for power=1.0 only. I want to add a test case for the strictly compound Poisson condition, that is 1 < power < 2: that is the "else:" branch in the code.

The `"mean_compound_poisson_deviance"` is not tested with non-NumPy array API inputs as far as I know.